File: LeetCode/Medium/2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.cpp
class Solution {
public:
    vector<int> findArray(vector<int>& pref) {
        int cur = 0;
        vector<int> ans;
        for(auto &num : pref){
            int elem = cur^num;
            ans.push_back(elem);
            cur = cur^elem;
        }
        return ans;
    }
};

File: LeetCode/Medium/2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.py
class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        # 101 000 -> 101
        cur = 0
        ans = []
        for num in pref:
            elem = cur^num
            ans.append(elem)
            cur = cur^elem
        return ans
        